### PR TITLE
Add resource config for lead controller resource

### PR DESCRIPTION
**Pull request by jackjlli**
This PR:
* adds resource config for lead controller resource to enable/disable the resource.
* explicitly enable lead controller resource, since the logic for this resource hasn't been merged yet.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
You want to add the field to enable/disable the resource right?

**Comment by jackjlli** [comment]
@Jackie-Jiang Yes, but we can also add the code saying if the field is missing or the field is `false`, then the resource is disabled by default.

**Comment by Jackie-Jiang** [comment]
You should still put the field when creating the resource config so it is more explicit, and easier for us to change

**Comment by jackjlli** [comment]
> You should still put the field when creating the resource config so it is more explicit, and easier for us to change

Done.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java [hunk 1]
```diff
@@ -177,5 +180,16 @@ private static void createLeadControllerResourceIfNeeded(String helixClusterName
...
       helixAdmin.addResource(helixClusterName, LEAD_CONTROLLER_RESOURCE_NAME, idealState);
     }
+
+    // Create resource config for lead controller resource if it doesn't exist
+    ResourceConfig resourceConfig = configAccessor.getResourceConfig(helixClusterName, LEAD_CONTROLLER_RESOURCE_NAME);
+    if (resourceConfig == null) {
+      resourceConfig = new ResourceConfig(LEAD_CONTROLLER_RESOURCE_NAME);
+    }
+    // Set RESOURCE_ENABLED to false if it's absent in resource config
+    if (resourceConfig.getSimpleConfig("RESOURCE_ENABLED") == null) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java]: putIfAbsent?

Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java]: I'd love to but `ResouceConfig` isn't a hashmap and there's no such method for now. :)

Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java]: Make sense lol

**Pull request merged**